core/node/auth/auth_impl.go

```diff
@@ -779,9 +800,53 @@ func (ca *chainAuth) checkEntitlement(
 		}
 	}
 
+	// Look for any returned errors. If at least one check was positive, then we ignore any subsequent
+	// errors. Otherwise we will report an error result since we could not conclusively determine that
+	// the user was not a space member.
+	for err := range isMemberError {
```

Comment:
reading errors will always wait for all goroutines to complete, even if isMember it true.